nucliadb_node/src/merge/scheduler.rs

```diff
@@ -41,38 +42,56 @@ use nucliadb_core::NodeResult;
 use super::work::WorkQueue;
 use crate::merge::{MergePriority, MergeRequest, MergeWaiter};
 
-const MAX_NODES_IN_MERGE: &str = "SCHEDULER_MAX_NODES_IN_MERGE";
-const SEGMENTS_BEFORE_MERGE: &str = "SCHEDULER_SEGMENTS_BEFORE_MERGE";
+const SCHEDULER_MAX_NODES_IN_MERGE: &str = "SCHEDULER_MAX_NODES_IN_MERGE";
+const SCHEDULER_SEGMENTS_BEFORE_MERGE: &str = "SCHEDULER_SEGMENTS_BEFORE_MERGE";
+const ON_COMMIT_MAX_NODES_IN_MERGE: &str = "MAX_NODES_IN_MERGE";
+const ON_COMMIT_SEGMENTS_BEFORE_MERGE: &str = "SEGMENTS_BEFORE_MERGE";
 
 /// Merge scheduler is the responsible for scheduling merges in the vectors
 /// index. When running, it takes the most prioritary merge request, takes a
 /// shard writer and executes the merge.
 pub struct MergeScheduler {
-    max_nodes_in_merge: usize,
-    segments_before_merge: usize,
     work_queue: Mutex<WorkQueue<InternalMergeRequest>>,
     shard_cache: Arc<ShardWriterCache>,
     settings: Settings,
     condvar: Condvar,
     shutdown: AtomicBool,
+    on_commit_parameters: MergeParameters,
+    scheduler_parameters: MergeParameters,
 }
 
 impl MergeScheduler {
     pub fn new(shard_cache: Arc<ShardWriterCache>, settings: Settings) -> Self {
-        Self {
-            shard_cache,
-            settings,
-            work_queue: Mutex::new(WorkQueue::new()),
-            condvar: Condvar::default(),
-            shutdown: AtomicBool::new(false),
-            max_nodes_in_merge: match std::env::var(MAX_NODES_IN_MERGE) {
+        let idle_parameters = MergeParameters {
+            max_nodes_in_merge: match std::env::var(SCHEDULER_MAX_NODES_IN_MERGE) {
                 Ok(v) => v.parse().unwrap_or(50_000),
                 Err(_) => 50_000,
             },
-            segments_before_merge: match std::env::var(SEGMENTS_BEFORE_MERGE) {
+            segments_before_merge: match std::env::var(SCHEDULER_SEGMENTS_BEFORE_MERGE) {
                 Ok(v) => v.parse().unwrap_or(2),
                 Err(_) => 2,
             },
+        };
+
+        let on_commit_parameters = MergeParameters {
+            max_nodes_in_merge: match std::env::var(ON_COMMIT_MAX_NODES_IN_MERGE) {
+                Ok(v) => v.parse().unwrap_or(50_000),
+                Err(_) => 50_000,
+            },
+            segments_before_merge: match std::env::var(ON_COMMIT_SEGMENTS_BEFORE_MERGE) {
+                Ok(v) => v.parse().unwrap_or(2),
+                Err(_) => 100,
+            },
+        };
```

Comment:
Why are we using env vars directly? Shouldn't we be using the `Settings` abstraction?

Reply:
Good point. I was just copying stuff around and missed the opportunity to move it into Settings. I'll change it, thanks! :sparkles: